hw1/tree.js

```diff
@@ -0,0 +1,66 @@
+/* Usage: node tree [.json file] */
+
+const args = process.argv.slice(2);
+
+if (args.length === 0) {
+	throw new Error('Please pass a .json file as an argument');
+}
+
+if (args.length > 1) {
+	throw new Error(`Too many arguments: expected 1, found ${args.length}`);
+}
+
+const data = JSON.parse(process.argv[2]);
+
+let dirs = 0;
+let files = 0;
+
+function recurse(data, depth = 0, unclosed = []) {
+	if (typeof data != 'object') {
+		throw new Error(`Error: expected "object" type, received ${typeof data}`);
+	}
+
+	if (!data.name) {
+		throw new Error('Error: no "name" property in the object');
+	}
+
+	let str = '';
+
+	for (let i = 0; i < depth; i++) {
+		if (i + 1 < depth) {
+			if (unclosed.some((x) => x == i)) {
```

Comment:
a little bit repeating your self on :37, maybe to have a separate function for it, also not clear from a reader pointer of view what's going on in code in this "if"